packages/fxa-settings/src/models/Account.ts

```diff
@@ -795,8 +795,10 @@ export class Account implements AccountData {
     cache.modify({
       id: cache.identify({ __typename: 'Account' }),
       fields: {
-        linkedAccounts: () => {
-          return [];
+        linkedAccounts: (existingAccounts) => {
```

Comment:
Some syntactic sugar could be
```
linkedAccounts: (existingAccounts) =>
  existingAccounts.filter((linkedAcc: LinkedAccount) => linkedAcc.providerId !== providerId))
```

But I'm fine with it being more explicit if that's preferred.